frontend/include/chpl/resolution/scope-types.h

```diff
@@ -478,12 +471,26 @@ class VisibilitySymbols {
     CONTENTS_EXCEPT,
   };
 
+  /** Indicating shadow scopes */
+  enum ShadowScope {
+    // public use / private import / public import all work at the same
+    // scope level as symbols declared within a module, and
+    // REGULAR_SCOPE indicates that.
+    REGULAR_SCOPE = 0,
+
+    // 'private use' aka just 'use' introduces 2 shadow scopes
+    // that appear to be between the module and any parent symbols
+    SHADOW_SCOPE_ONE = 1, // for module contents brought in by a 'private use'
+    SHADOW_SCOPE_TWO = 2, // for module name from a 'private use'
+  };
+
  private:
   const Scope* scope_; // Scope of the Module etc
                        // This could technically be an ID but basically
                        // anything we do with it needs a Scope* anyway.
   Kind kind_ = SYMBOL_ONLY;
   bool isPrivate_ = true;
+  int8_t shadowScopeLevel_ = REGULAR_SCOPE;
```

Comment:
It saves 4 bytes of memory in each instance here due to the way that C / C++ have enums be `int` and the way that fields are aligned. If it is a 1 byte type, it can be stored next to the `bool`, instead of on the next 4-byte aligned word.

We could apply this space optimization in many other places in the uAST but somehow it seemed more important to me in this case, where the enum only stores 3 values. Thanks to our efforts at encapsulation, this field is private, so the type of it is not particularly important to most of the code.